modules/reactor/src/meshgenerators/PolygonConcentricCircleMeshGeneratorBase.C

```diff
@@ -871,6 +886,78 @@ PolygonConcentricCircleMeshGeneratorBase::generate()
           i + interface_id_shift + _interface_boundary_id_shift) = _interface_boundary_names[i];
     }
   }
+  // sector id
+  if (getParam<bool>("use_sector_id"))
+  {
+    const auto extra_id_index = mesh0->add_elem_integer(getParam<std::string>("sector_id_name"));
+    std::vector<dof_id_type> elem_sector_ids(mesh0->n_elem(), 0);
+    unsigned int id = 1;
+    unsigned int offset = 0;
+    for (unsigned int is = 0; is < _num_sides; ++is)
```

Comment:
why do it now after the slice has been made, duplicated and stitched, instead of while building the slice ?

Reply:
also will you never need this for the other classes using the slice utility?

I cant remember, is the TriPinMG using PCCMG or is it using the slices?

Reply:
For your first comment:
I intentionally did it - making the ID after the slice creation.
That was because I did not want to make the slice creation process more complicated.
Doing after the slice creation is less efficient but I think it would be simpler in terms of maintenance.